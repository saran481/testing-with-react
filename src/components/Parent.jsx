import React, { Component } from 'react';
import ChildComponent from './ChildComponent';

export default class Parent extends Component {
  constructor (props) {
       super(props);
       this.state={
         saran : "hey there"
       }
      this.label = "I’m your father";
      
  };

  onSubmit = (option) => (option === "I’m your father" ? 'cool' : 'not cool');

  render () {
    return(
      <div className="container">
        <ChildComponent label={this.label} onSubmit={this.onSubmit} />
      </div>
    );
  };
};