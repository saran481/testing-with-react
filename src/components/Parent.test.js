import React from 'react';
import { shallow } from 'enzyme';
import renderer from 'react-test-renderer'
import Parent from './Parent'

jest.mock('./childComponent.jsx')

let wrapper = shallow(<Parent />);

var container, containerProp,childContainerProps;

var childContainer

describe('parent component snapshot', () => {
 it('matches the snapshot', () => {
     const tree =renderer.create(<Parent />).toJSON()
      expect(tree).toMatchSnapshot() 
 })

})

describe("Parent", ()=> {
   beforeEach(()=>{
       container = wrapper.find(".container")
       containerProp=container.props();
   })
})

it("should have div", () => {
    expect(wrapper.find(".container")).toHaveLength(1);
});

describe('childcomponent', () => {
    beforeEach(() =>{
        childContainer = wrapper.find("ChildComponent");
       childContainerProps = childContainer.props();
    })
})

it('shoudl find child in parent component', () => {
    expect(wrapper.find("ChildComponent")).toHaveLength(1)
})

it('access child container props', () => {
    expect(wrapper.find("ChildComponent").props().label).toEqual('Iâ€™m your father')
})

it('accessing state variables', () => {
    // console.log(wrapper.state())
      // const saran= "saran"
      expect(wrapper.state().saran).toBe("hey there")
})

