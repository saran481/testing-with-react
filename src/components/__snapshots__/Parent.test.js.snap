// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`parent component snapshot matches the snapshot 1`] = `
<div
  className="container"
>
  <div
    className="container"
  >
    <input
      onClick={[Function]}
      type="button"
    >
      Iâ€™m your father
    </input>
  </div>
</div>
`;
